Manual/Iterators.lean

```diff
@@ -342,6 +369,44 @@ This iterator is useful with combinators such as {name}`Iter.zip`:
 2: pachycephalosaurus
 ```
 :::
+
+:::paragraph
+In contrast to the previous example, this loop terminates because `xs.iter` is a finite iterator,
+One can make sure that a loop actually terminates by providing a {name}`Finite` instance:
+```lean (name := natfin)
+#check type_of% (Nats.iter.zip ["cat", "dog"].iter).internalState
```

Comment:
This is a nice trick! We don't describe `type_of%` anywhere yet. Should we think of it as a public user-facing feature? Maybe we should add an option to `#check` that suppresses the term, kind of like how `#reduce` has options to reduce in types/proofs?